src/main/java/io/jenkins/docker/client/DockerMultiplexedInputStream.java

```diff
@@ -54,7 +54,8 @@ private void readInternal() throws IOException {
                     byte[] payload = new byte[size];
                     int received = 0;
                     while (received < size) {
-                        i = multiplexed.read(payload, received, size - received);
+                        int i = multiplexed.read(payload, received, size - received);
+                        if (i < 0) return; // EOF
                         received += i;
                     }
                     System.err.println(new String(payload, StandardCharsets.UTF_8));
```

Comment:
would indeed make perfect sense

Reply:
@a-ba now we have a `break` in the `while` loop, we now have a scenario where we might call `new String(payload, StandardCharsets.UTF_8)` where payload is incompletely populated with data.
We'll have to use `new String(payload, 0, received, StandardCharsets.UTF_8)` instead.

At this point, I think we may as well fix both issues with this code by adding `import org.slf4j.Logger;` and `import org.slf4j.LoggerFactory;`, defining a `private static final Logger LOGGER = LoggerFactory.getLogger(DockerMultiplexedInputStream.class);` (at the start of the class) and then, instead of calling `System.err.println(String)`, we should call `LOGGER.warn(String, Object...)` so the stderr text gets logged to a Logger instead of just the Jenkins server's stderr.

e.g. something like
```
final String stderr = new String(payload, 0, received, StandardCharsets.UTF_8);
if (received==size) {
  LOGGER.warn("Unexpected stderr on stream: {}", stderr);
} else {
  LOGGER.warn("Unexpected stderr on stream, EOF after {} (of {}) bytes received: {}", received, size, stderr);
}
```

Reply:
fixed in 5655f86c689fa66087cce757e204622a96b99405